### PR TITLE
fix(client): return an empty string for empty sections

### PR DESCRIPTION
**Pull request by ValeraS**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.

### CODE REVIEW

File: tools/challenge-md-parser/text-to-data.js [hunk 1]
```diff
@@ -62,7 +62,12 @@ function textToData(sectionIds) {
               }
             }
           });
-          const textArray = toHTML({ ...node, children: newChildren });
+          const hasData = newChildren.some(
+            node => node.type !== 'text' || !/^\s*$/.test(node.value)
+          );
+          const textArray = hasData
```
Comment (Bouncey) [inline tools/challenge-md-parser/text-to-data.js]: Can you add a test to cover this?

**Review by Bouncey — CHANGES_REQUESTED** [review]

File: tools/challenge-md-parser/text-to-data.js (hunk 1 above)
Comment (ValeraS) [reply tools/challenge-md-parser/text-to-data.js]: done, is it ok?

**Comment by Bouncey** [comment]
@ValeraS I have added some commits to the PR.

- Undo changes to the mock ast and the snapshot
- Add a second ast with no instructions for use in the new test

This ensures that we are still testing the output for a challenge with instructions and without.

Let me know if these changes are ok with you, then I can merge.

**Review by Bouncey — APPROVED** [review]

**Comment by ValeraS** [comment]
It's ok, but I don't see the need. For these tests, it does no matter which section names we use. So we have tests for:
- contentful section
- empty section
- and the missing section
and they might not depend on challenge sections.

**Comment by Bouncey** [comment]
You are right, yes. I failed to read the rest of the tests.

**Pull request merged**